block/manager.go

```diff
@@ -61,10 +65,21 @@ type Manager struct {
 	// daHeight is the height of the latest processed DA block
 	daHeight uint64
 
-	HeaderCh  chan *types.SignedHeader
-	BlockCh   chan *types.Block
-	blockInCh chan newBlockEvent
-	syncCache map[uint64]*types.Block
+	HeaderCh chan *types.SignedHeader
+	BlockCh  chan *types.Block
+
+	blockInCh  chan newBlockEvent
+	blockStore *goheaderstore.Store[*types.Block]
+
+	syncCache              map[uint64]*types.Block
+	isBlockWithHashSeenMtx *sync.RWMutex
+	isBlockWithHashSeen    map[string]bool
+	isBlockHardConfirmed   map[string]bool
```

Comment:
I think we could/should group these all under the same mutex. These are all caches right? So it would seem that we would want these all to be synced right? 

then it would be something like:
```go
type blockCache struct {
  blocks    map[uint64]*types.Block
  hashes   map[string]bool
  hardConfirmations  map[string]bool
  mtx   *sync.RWMutex
}

func (bc *blockCache) block(unit64) (bool, block) {
   bc.mtx.Lock()
  defer bc.mtx.Unlock()
  return bc.blocks[uint64]
}

func (bc *blockCache) isSeen(string) bool {
   bc.mtx.Lock()
  defer bc.mtx.Unlock()
  okHash, _ := bc.hashes[string]
  okHC, _ := bc.hardConfirmations[string]
  return okHash || okHC
}
```

Reply:
Great suggestion. Created a separate file for the new struct and related methods.